modular/gater/request_context.go

```diff
@@ -266,6 +307,39 @@ func (r *RequestContext) verifyOffChainSignature(requestSignature string) (sdk.A
 	}
 }
 
+// verifyOffChainSignatureFromPreSignedURL used to verify off-chain-auth signature, return (address, nil) if check succeed.  The auth information will be parsed from URL.
+func (r *RequestContext) verifyOffChainSignatureFromPreSignedURL(authenticationStr string, account string, domain string) (sdk.AccAddress, error) {
+	var (
+		signedMsg *string
+		err       error
+	)
+	signedMsg, sigString, err := parseSignedMsgAndSigFromRequest(authenticationStr)
+	if err != nil {
+		return nil, err
+	}
+
+	// check request integrity
+	realMsgToSign := commonhttp.GetMsgToSignForPreSignedURL(r.request)
+	if hex.EncodeToString(realMsgToSign) != *signedMsg {
+		log.CtxErrorw(r.ctx, "failed to check signed msg")
+		return nil, ErrRequestConsistent
+	}
+
+	offChainSig := *sigString
+
+	verifyOffChainSignatureResp, err := r.g.baseApp.GfSpClient().VerifyOffChainSignature(r.Context(), account, domain, offChainSig, realMsgToSign)
+	if err != nil {
+		log.Errorf("failed to verify off chain signature", "error", err)
+		return nil, err
+	}
+	if verifyOffChainSignatureResp {
+		userAddress, _ := sdk.AccAddressFromHexUnsafe(account)
+		return userAddress, nil
+	} else {
+		return nil, err
+	}
+}
+
 func parseSignedMsgAndSigFromRequest(requestSignature string) (*string, *string, error) {
```

Comment:
As we discussed,  `requestSignature` doesn't contain any `SignedMsg`, to make the header simple?

Reply:
Done